src/main/scala/edu/alexy/lectures/concurrency/RoboAdviser.scala

```diff
@@ -2,50 +2,95 @@ package edu.alexy.lectures.concurrency
 
 import edu.alexy.lectures.concurrency.model.Company
 import edu.alexy.lectures.concurrency.util.{Db, DbError, WebApi, WebApiError}
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
+import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+
 object RoboAdviser {
   // Task 1.
   // Return 'AAPL' revenue from `Db.getCompanyLastFinancials`. Possible error should be returned as a ServiceError.
-  def getAAPLRevenue: Future[Double] = ???
+  def getAAPLRevenue: Future[Double] = Db.getCompanyLastFinancials("AAPL").transform {
+    case Success(value) => Success(value.map(_.revenue.toDouble).getOrElse(0.0))
+    case Failure(_) => Failure(DbError)
+  }
 
   // Task 2.
   // Implement a fallback strategy for 'Db.getAllTickers'.
   // 'Db.getAllTickers' should be called no more then 'retries' times.
   // Possible error should be returned as a ServiceError.
-  def getAllTickersRetryable(retries: Int = 10): Future[Seq[String]] = ???
+  def getAllTickersRetryable(retries: Int = 10): Future[Seq[String]] = {
+    val res = Db.getAllTickers
+    if (retries <= 0) res.transform {
```

Comment:
With such a logic `getAllTickersRetryable` could be could `retries + 1` times but no more than `retries` times is expected.

Reply:
Forget about my comment above. That's ok. My mistake.